### PR TITLE
Seat equip notif init

### PR DESCRIPTION
**Pull request by jakhongirzoyirov**

### CODE REVIEW

File: src/main/resources/db.changelog/changes/em_no.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+databaseChangeLog:
+  - logicalFilePath: no-path
+  - changeSet:
+      id: em_no table init
```
Comment (ikantsevich) [inline src/main/resources/db.changelog/changes/em_no.yaml]: em_no is a bad name for table

File: src/main/resources/db.changelog/changes/em_no.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+              - column:
+                  name: no_id
+                  type: int
+                  constraints:
+                    unique: true
+                    nullable: true
+                    foreignKeyName: fk_no
+                    references: no(no_id)
+              - column:
+                  name: em_no_created
```
Comment (ikantsevich) [inline src/main/resources/db.changelog/changes/em_no.yaml]: what about modified?

File: src/main/resources/db.changelog/changes/em_no.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+databaseChangeLog:
+  - logicalFilePath: no-path
+  - changeSet:
+      id: em_no table init
+      author: Jakhongir
+      preConditions:
```
Comment (ikantsevich) [inline src/main/resources/db.changelog/changes/em_no.yaml]: where is primary key?

File: src/main/resources/db.changelog/changes/em_no.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+                  name: no_id
+                  type: int
+                  constraints:
+                    unique: true
+                    nullable: true
+                    foreignKeyName: fk_no
+                    references: no(no_id)
+              - column:
+                  name: em_no_created
+                  type: CURRENT_TIMESTAMP
```
Comment (ikantsevich) [inline src/main/resources/db.changelog/changes/em_no.yaml]: Type CURRENT_TIMESTAMP? Do these scripts work ok?

File: src/main/resources/db.changelog/changes/notification.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+                  name: at_id
+                  type: int
+                  constraints:
+                    unique: true
+                    nullable: true
+                    foreignKeyName: fk_at
+                    references: at(at_id)
+              - column:
+                  name: message
+                  type: varchar
```
Comment (ikantsevich) [inline src/main/resources/db.changelog/changes/notification.yaml]: just varchar?

File: src/main/resources/db.changelog/changes/notification.yaml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+              - column:
+                  name: at_id
+                  type: int
+                  constraints:
+                    unique: true
+                    nullable: true
+                    foreignKeyName: fk_at
+                    references: at(at_id)
+              - column:
+                  name: message
```
Comment (ikantsevich) [inline src/main/resources/db.changelog/changes/notification.yaml:32]: What about prefix?

**Review by ikantsevich — DISMISSED** [review]

File: src/main/resources/db.changelog/changes/em_no.yaml (hunk 4 above)
Comment (jakhongirzoyirov) [reply src/main/resources/db.changelog/changes/em_no.yaml]: ok so, i should make it datetime, right?

File: src/main/resources/db.changelog/changes/notification.yaml (hunk 6 above)
Comment (jakhongirzoyirov) [reply src/main/resources/db.changelog/changes/notification.yaml:32]: could you make it clarify pliz?

File: src/main/java/com/exadel/sandbox/em_notif/service/impl/EmployeeNotificationServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+        List<EmployeeNotification> employeeNotificationList = emplyeeNotificationRepository.findAll();
+
+        List<EmployeeNotificationResponseDto> employeeNotificationResponseDtos = new ArrayList<>();
+
+        for (EmployeeNotification emplNotif:
+             employeeNotificationList) {
+            employeeNotificationResponseDtos.add(fullMap(emplNotif));
+        }
+
```
Comment (KirvelPasha) [inline src/main/java/com/exadel/sandbox/em_notif/service/impl/EmployeeNotificationServiceImpl.java]: Please  write this method something like that: 

List<EmployeeNotification> employeeNotificationList = emplyeeNotificationRepository.findAll();

        return employeeNotificationList.stream()
                .map(this::fullMap)
                .collect(Collectors.toList());

Comment (KirvelPasha) [reply src/main/java/com/exadel/sandbox/em_notif/service/impl/EmployeeNotificationServiceImpl.java]: Also please check other places and try to use stream where possible

File: src/main/java/com/exadel/sandbox/em_notif/service/impl/EmployeeNotificationServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    public EmployeeNotificationResponseDto getEmNotifById(Long id) {
+
+        Optional<EmployeeNotification> byId = emplyeeNotificationRepository.findById(id);
+
+        EmployeeNotification employeeNotification = byId.orElseThrow( ()-> new EntityNotFoundException("employee_notification with id: " + id + " not found"));
+
+
+        return fullMap(employeeNotification);
+
+
```
Comment (KirvelPasha) [inline src/main/java/com/exadel/sandbox/em_notif/service/impl/EmployeeNotificationServiceImpl.java]: Please remove extra empty lines. Apply reformat code for project

File: src/main/java/com/exadel/sandbox/notification/service/impl/NotificationServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+        return fullMap(notificationRepository.save(notification));
+    }
+
+
+
+    private NotificationResponseDto fullMap(Notification notification) {
+        NotificationResponseDto notificationResponseDto = mapper.map(notification, NotificationResponseDto.class);
+
+
+        return null;
```
Comment (KirvelPasha) [inline src/main/java/com/exadel/sandbox/notification/service/impl/NotificationServiceImpl.java]: Please check this method

File: src/main/java/com/exadel/sandbox/em_notif/service/impl/EmployeeNotificationServiceImpl.java (hunk 8 above)
Comment (jakhongirzoyirov) [reply src/main/java/com/exadel/sandbox/em_notif/service/impl/EmployeeNotificationServiceImpl.java]: thanks but these parts are not my code, I wrote codes for seat and equip tables only

File: src/main/resources/db.changelog/db.changelog-master.yaml [hunk 10]
```diff
@@ -5,6 +5,22 @@ databaseChangeLog:
...
+      relativeToChangelogFile: true
+
+  - include:
+      file: changes/notification.yaml
+      relativeToChangelogFile: true
+
+  - include:
+      file: changes/em_notif.yaml
+      relativeToChangelogFile: true
+
```
Comment (ghost) [inline src/main/resources/db.changelog/db.changelog-master.yaml]: I think you should move these lines to the end. Please we may have some conflicts when creating table.

File: src/main/resources/db.changelog/changes/seat.yaml [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+              - column:
+                  name: fl_id
+                  type: int
+                  constraints:
+                    unique: true
+                    nullable: true
+                    foreignKeyName: fk_fl
+                    references: fl(fl_id)
+              - column:
+                  name: seat_num
```
Comment (ghost) [inline src/main/resources/db.changelog/changes/seat.yaml:32]: can you add constraints not null for these?

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Review by jakhongirzoyirov — COMMENTED** [review]
viewed

**Review by jakhongirzoyirov — COMMENTED** [review]
fixed

File: src/main/resources/db.changelog/db.changelog-master.yaml [hunk 12]
```diff
@@ -16,8 +24,6 @@ databaseChangeLog:
   - include:
       file: changes/vacation.yaml
       relativeToChangelogFile: true
-      relativeToChangelogFile: true
```
Comment (believer-0910) [inline src/main/resources/db.changelog/db.changelog-master.yaml:19]: here you should delete extra row

File: src/main/resources/db.changelog/db.changelog-master.yaml [hunk 13]
```diff
@@ -5,6 +5,14 @@ databaseChangeLog:
...
       relativeToChangelogFile: true
 
+  - include:
+      file: changes/seat.yaml
+      relativeToChangelogFile: true
+
+  - include:
+      file: changes/equipment.yaml
+      relativeToChangelogFile: true
+
```
Comment (believer-0910) [inline src/main/resources/db.changelog/db.changelog-master.yaml:15]: please add yor changes to the end of the changelog file

**Review by believer-0910 — APPROVED** [review]

**Pull request merged**